hazelcast/src/main/java/com/hazelcast/sql/impl/expression/RowValue.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A value of a ROW type (ROW is a SQL struct). It contains values, but not the
+ * field names - names are part of the type, the value doesn't reference the
+ * type.
+ */
+public class RowValue implements Serializable, IdentifiedDataSerializable {
```

Comment:
Are we returning this to user? I see that it is added as a column type, but this class is in the impl package, which is not a place for the public API.

Reply:
No, we're converting it to a String essentially for now. Its converted in the Codec. I'm not sure what's the ideal solution here actually, maybe we can revisit this during beta? I think `TO_ROW` function (which is the only place where its used) might just return VARCHAR instead. Not sure what's the best way to return value actually, TO_ROW is meant for cases where client doesn't have target Class in the classpath (or when its not even a Java client).

Reply:
I see, then I think we can re-visit that in the BETA :+1: